### ISSUE TITLE
Is there a way to make it chat in a server instead of dms?

### ISSUE THREAD

**Issue by Kitteh32842** [enterv0id/ChatGPT-Discord-Selfbot#1]
That would be really cool, fr

**Comment by edwardelton** [comment]
Yes you can. Check if the message.guild id is the guild id from the server you want to listen on. It will work fine after.

**Comment by sallonium** [comment]
> Yes you can. Check if the message.guild id is the guild id from the server you want to listen on. It will work fine after.

May I beg your pardon where can I find that ? (message.guild)

**Comment by edwardelton** [comment]
> > Yes you can. Check if the message.guild id is the guild id from the server you want to listen on. It will work fine after.
> 
> May I beg your pardon where can I find that ? (message.guild)

if message.guild is None and message.author != client.user:
Basically, if the message is not from a guild (server) and the author is not its self.

line 60, you can add server by replacing it with this : 

```py
if message.author != client.user:
```

You will probably get rate limited by open AI api tho.
You can think of a way to not get rate limit or make the AI wait and not reply to everything messages sent in the server.

**Comment by sallonium** [comment]
Hello, 
In order to not get rate limited by openAI's API I've came up with this way where users must ping the bot (or me lol) in order to interact with it;
I've divided my work into two:
When they must ping for each assistance:
`@client.event
async def on_message(message):
    if message.author == client.user:
        return

    # Check if the message mentions the bot
    if client.user in message.mentions:
        # Remove the bot's mention from the message content
        content = message.content.replace(client.user.mention, '')

        # Call the OpenAI API to generate a response
        response = openai.Completion.create(
            engine='davinci-codex',
            prompt=content,
            max_tokens=50,
            n=1,
            stop=None,
            temperature=0.7
        )

        # Get the generated response from OpenAI
        reply = response.choices[0].text.strip()

        # Send the response back to the Discord channel
        await message.channel.send(reply)`
But that'll probably cause chaos and mass mention so why not just ping once to open a conversation, sadly I thought about this once I was done with the code above so..just wanted to share it anyway so I don't feel shit..
**Here's when they must ping *once* for a full convo:**
`@client.event
async def on_message(message):
    if message.author == client.user:
        retur

    if client.user in message.mentions:
        user_id = message.author.id

        # Start a new conversation if user is not in an existing conversation
        if user_id not in conversations:
            conversations[user_id] = {'messages': []}

        # Add the user's message to the conversation
        conversations[user_id]['messages'].append(message.content)

        # Call the OpenAI API to generate a response
        response = openai.Completion.create(
            engine='davinci-codex',
            messages=conversations[user_id]['messages'],
            max_tokens=50,
            n=1,
            stop=None,
            temperature=0.7
        )

        # Get the generated response from OpenAI
        reply = response.choices[0].message.content.strip()

        # Add the bot's reply to the conversation
        conversations[user_id]['messages'].append(reply)

        # Send the response back to the Discord channel
        await message.channel.send(reply)
`
I hope that you let me know if I made a mistake or some sort of it. 
Thanks,

**Comment by edwardelton** [comment]
From what I see, your code should work!

Great idea for the ping technique.

**Comment by KansaiDorito** [comment]
Hi, the code posted above isn't working, keeps saying that the "conversations" variable is undefined, any help?
![image](https://github.com/enterv0id/ChatGPT-Discord-Selfbot/assets/99162098/acf28fc7-9d0a-46d1-89c6-c4acb57ede56)

**Comment by edwardelton** [comment]
It literally tell you the variable does not exist.

**Comment by KansaiDorito** [comment]
Alright, forgive my stupidity, then. But what can I do to remedy this, and allow the selfbot to generate responses in a server?

**Comment by edwardelton** [comment]
> Alright, forgive my stupidity, then. But what can I do to remedy this, and allow the selfbot to generate responses in a server?

> > > Yes you can. Check if the message.guild id is the guild id from the server you want to listen on. It will work fine after.
> > 
> > 
> > May I beg your pardon where can I find that ? (message.guild)
> 
> ```python
> if message.guild is None and message.author != client.user:
> ```
> 
> Basically, if the message is not from a guild (server) and the author is not its self.
> 
> line 60, you can add server by replacing it with this :
> 
> ```python
> if message.author != client.user:
> ```
> 
> You will probably get rate limited by open AI api tho. You can think of a way to not get rate limit or make the AI wait and not reply to everything messages sent in the server.

Check above, I wrote this.

[sallonium](https://github.com/sallonium) Posted a way that the bot only answer when pinged like Clyde, the AI discord.

You can also check my repo where there is an example where the bot keeps tracking of the conversation, etc.
